src/main/java/com/example/gymbuddy/implementation/controllers/MachineHistoryController.java

```diff
@@ -0,0 +1,28 @@
+package com.example.gymbuddy.implementation.controllers;
+
+import com.example.gymbuddy.infrastructure.models.dtos.MachineHistoryDto;
+import com.example.gymbuddy.infrastructure.services.MachineHistoryService;
+import com.example.gymbuddy.infrastructure.services.MachineService;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/machine_history")
```

Comment:
One thing we want to consider with this path is how do we pass in data to get history of a member and a specific machine.

`/machine/{machine_id}/history/{member_id}` is one solution, but is this the best way to do it?

This isn't well defined because this history is part of a member **and** a machine,

Nomenclature for endpoints is pretty standared,

so when I see `/machine/{machine_id}` to me that means we are retrieving a machine.

`/machine/{machine_id}/history` would indicate to me that we are getting all the history of a specific machine

`member/{member_id}/history` would indicate to me that we are getting all the history of a specific member

So the issue is, I'm not sure the best way to get data for a member and a machine. I'm feeling something like

`/member/{member_id}/machine/{machine_id}/history`